manipulate_pim_data/mass_edition/register_a_new_bulk_action.rst

```diff
@@ -0,0 +1,280 @@
+                - '@akeneo_batch.job_repository'
+                - '@pim_enrich.reader.database.product_and_product_model'
+                - '@acme.connector.processor.mass_edit.product.add_comment'
+                - '@pim_connector.writer.database.product'
+
+.. note::
+
+    Each job needs to define a default value provider and a constraint collection provider.
+    In this very simple case, we will use the default product mass edit ones, so we added it without any additional parameter.
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/AppBundle/Resources/config/jobs.yml
+    services:
+        acme.job.add_comment:
+            class: '%pim_connector.job.simple_job.class%'
+            arguments:
+                - 'add_comment'
+                - '@event_dispatcher'
+                - '@akeneo_batch.job_repository'
+                - [ '@acme.step.add_comment.mass_edit' ]
+            tags:
+                -
+                    name: akeneo_batch.job
+                    connector: '%pim_enrich.connector_name.mass_edit%'
+                    type: '%pim_enrich.job.mass_edit_type%'
+
+        acme.job.default_values_provider.add_comment:
+            class: '%pim_enrich.connector.job.job_parameters.default_values_provider.product_mass_edit.class%'
+            arguments:
+                - [ 'add_comment' ]
+            tags:
+                - { name: akeneo_batch.job.job_parameters.default_values_provider }
+
+        acme.job.constraint_collection_provider.add_comment:
+            class: '%pim_enrich.connector.job.job_parameters.constraint_collection_provider.product_mass_edit.class%'
+            arguments:
+                - [ 'add_comment' ]
+            tags:
+                - { name: akeneo_batch.job.job_parameters.constraint_collection_provider }
+
+As we've done previously, add these files to the dependency injection:
+
+.. code-block:: php
+
+    // src/Acme/Bundle/AppBundle/DependencyInjection/AcmeAppExtension.php
+    [...]
+    $loader->load('jobs.yml');
+    $loader->load('steps.yml');
+    [...]
+
+Finally, add this new job instance in the database to be able to run it:
+
+.. code-block:: bash
+
+    bin/console akeneo:batch:create-job internal add_comment mass_edit add_comment '{}' 'Add comment' --env=prod
+
+Create the UI
+-------------
+
+The UI of this mass action is simple: we just have to create a textarea field.
+When user updates this textarea, the data will be put in the form data.
+
+.. code-block:: html
+
+    <!-- src/Acme/Bundle/AppBundle/Resources/public/templates/add-comment.html -->
+    <div class="AknFieldContainer">
+        <div class="AknFieldContainer-header">
+            <label class="AknFieldContainer-label">Comment</label>
+        </div>
+        <div class="AknFieldContainer-inputContainer">
+            <textarea class="AknTextareaField comment-field" <% if (readOnly) { %> disabled="disabled"<% } %>><%- value %></textarea>
+        </div>
+    </div>
+
+.. code-block:: javascript
```

Comment:
And here is the javascript ..... explain a bit here what are those two components prior to show the code

Reply:
I did it on the begin of the subsection